Source/WebCore/crypto/mac/CryptoAlgorithmEd25519Mac.cpp

```diff
@@ -0,0 +1,76 @@
+//
+//  CryptoAlgorithmEd25519.cpp
+//  WebCore
+//
+//  Created by Ángela Izquierdo García on 14/9/22.
+//
+
+#include <stdio.h>
+ /* CryptoAlgorithmEd25519 */
+#include "config.h"
+#include "CryptoAlgorithmEd25519.h"
+
+#if ENABLE(WEB_CRYPTO)
+#include "CommonCryptoUtilities.h"
+#include "CryptoKeyEC.h"
+#include "CryptoAlgorithmEd25519Params.h"
+#include <pal/spi/cocoa/CoreCryptoSPI.h>
+
+#include <string.h>
+
+
+#include "CommonCryptoDERUtilities.h"
+
+#include "CryptoDigestAlgorithm.h"
+#include <wtf/text/Base64.h>
+#include <wtf/Assertions.h>
+namespace WebCore {
+static ExceptionOr<Vector<uint8_t>> signEd25519(CryptoAlgorithmIdentifier hash, const PlatformECKey key, size_t keyLengthInBytes, const Vector<uint8_t>& data)
+{
+    (void) hash;
+    (void) keyLengthInBytes;
+    (void) key;
+    Vector<uint8_t> sk = std::get<Vector<uint8_t>>(key);
+    ccec25519pubkey pk;
+    const struct ccdigest_info *di = ccsha512_di();
+    cced25519_make_pub(di,  pk,  sk.data());
+    Vector<uint8_t> publicKey(pk,32);
+    ccec25519signature newSignature = {} ;
+    cced25519_sign(di,newSignature,data.size(),data.data(),pk,sk.data());
+    Vector<uint8_t> sg(newSignature,64);
+    return WTFMove( sg);
+
+}
+
+static ExceptionOr<bool> verifyEd25519(CryptoAlgorithmIdentifier hash, const PlatformECKey key, size_t keyLengthInBytes, const Vector<uint8_t>& signature, const Vector<uint8_t> data)
+{
+    (void) keyLengthInBytes;
+    (void) hash;
+    (void) key;
```

Comment:
As they are parameters not used passed to the function it was giving a compiling error but the parameters passed should remain the same so that's why I wrote it that way